usaspending_api/awards/v1/views.py

```diff
@@ -182,3 +185,57 @@ def get_queryset(self):
         filtered_queryset = self.filter_records(self.request, queryset=queryset)
         ordered_queryset = self.order_records(self.request, queryset=filtered_queryset)
         return ordered_queryset
+
+
+class FABAFilter(filters.FilterSet):
+
+    class Meta:
+        model = FinancialAccountsByAwards
+        # You can create any relationships you want here using ORM notation
+        fields = ('piid', 'treasury_account__agency_id')
+
+
+class FinancialAccountsByAwardsListViewset(viewsets.ReadOnlyModelViewSet):
+
+    '''
+        I spent a lot of time investigating the serializers here trying to come up with a better solution. However
+        ultimately I determined that the LimitableSerializer class which we base all our serializers on is actually
+        fairly well put together. It handles prefetching well, so that the nested serializers don't prompt new queries,
+        and it doesn't really mess with the core functionality of the serializer as written in DRF. My main objection
+        to them is having to specify default_fields unless verbose is set. If I was ultimately going to rewrite this,
+        I might switch that around so that verbose is the default. I did not feel that that was worth investing time in
+        for this investigation however.
+    '''
+
+    serializer_class = FinancialAccountsByAwardsSerializer
```

Comment:
Once added, I was hoping to see the ability to filter on a related field, who was added to the serializer class and given an alias. See:
https://www.django-rest-framework.org/api-guide/serializers/#specifying-fields-explicitly
or 
https://www.django-rest-framework.org/api-guide/relations/#primarykeyrelatedfield
```
class AlbumSerializer(serializers.ModelSerializer):
    tracks = serializers.PrimaryKeyRelatedField(many=True, read_only=True)

    class Meta:
        model = Album
        fields = ('album_name', 'artist', 'tracks')
```

In your case, maybe use something like that to wrap around the `treasury_account__agency_id` relation and have it returned as a field in the response just named `agency`.